protocol/flows/src/flow_context.rs

```diff
@@ -106,6 +154,80 @@ impl FlowContext {
     pub async fn unorphan_blocks(&self, root: Hash) -> Vec<Block> {
         self.orphans_pool.write().await.unorphan_blocks(root).await
     }
+
+    /// Updates the mempool after a new block arrival, relays newly unorphaned transactions
+    /// and possibly rebroadcast manually added transactions when not in IBD.
+    ///
+    /// _GO-KASPAD: OnNewBlock + broadcastTransactionsAfterBlockAdded_
+    pub async fn on_new_block(&self, block: Block) -> Result<(), ProtocolError> {
+        let hash = block.hash();
+        let blocks = self.unorphan_blocks(hash).await;
+        // Use a ProcessQueue so we get rid of duplicates
+        let mut transactions_to_broadcast = ProcessQueue::new();
+        for block in once(block).chain(blocks.into_iter()) {
+            transactions_to_broadcast
+                .enqueue_chunk(self.mining_manager().handle_new_block_transactions(&block.transactions)?.iter().map(|x| x.id()));
+        }
+
+        // Don't relay transactions when in IBD
+        if self.is_ibd_running() {
```

Comment:
I guess it should be `is_nearly_synced` here as well

Reply:
In original go codebase, [`is_ibd_running()` is used](https://github.com/tiram88/kaspad/blob/bd1420220a1c9f7ab253b2b120240351e9440146/app/protocol/flowcontext/blocks.go#L73).

Perhaps, yes, we might be more permissive and use `is_nearly_synced()`.

Reply:
so we changed this in the main tx relay flow, but missed this one